code/game/machinery/firealarm.dm

```diff
@@ -499,25 +612,23 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/firealarm, 26)
 	desc = "Cuban Pete is in the house!"
 	var/static/party_overlay
 
-/obj/machinery/firealarm/partyalarm/reset()
-	if (machine_stat & (NOPOWER|BROKEN))
+/obj/machinery/firealarm/partyalarm/reset(mob/user, silent = FALSE)
+	if (!is_operational || !can_reset)
 		return
 	var/area/area = get_area(src)
 	if (!area || !area.party)
```

Comment:
in `/obj/machinery/firealarm/partyalarm/alarm(...)` you replaced the `get_area(src)` and `!area` checks with `my_area`
is this not using it intentional?